### PR TITLE
Add better network detection

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR helps improve network detection by trying to connect to the client server to test the connection before reporting an online/offline state.

### CODE REVIEW

**Comment by enahum** [comment]
@jarredwitt this one is failiing

File: app/utils/network.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    };
+}
+
+export default function networkConnectionListener(onChange) {
+    const connectionChanged = handleConnectionChange(onChange);
+
+    NetInfo.isConnected.addEventListener('change', connectionChanged);
+    NetInfo.isConnected.fetch().then(connectionChanged);
+
+    const removeEventListener = () => NetInfo.isConnected.removeEventListener('change', connectionChanged); // eslint-disable-line
```
Comment (enahum) [inline app/utils/network.js]: why the eslint disable line here?

**Comment by hmhealey** [comment]
This appears to be causing issues with the unit tests
```
1) Actions.Views.Login "before each" hook: _callee for "handleLoginIdChanged":
     TypeError: Cannot read property 'isConnected' of undefined
      at networkConnectionListener (app/utils/network.js:34:2)
      at Object.detectNetwork (app/store/index.js:199:2)
      at configureServiceStore (node_modules/mattermost-redux/store/configureStore.dev.js:91:27)
      at configureAppStore (app/store/index.js:199:2)
      at Object._callee$ (test/app/actions/login.test.js:14:17)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at invoke (node_modules/regenerator-runtime/runtime.js:155:20)
      at node_modules/regenerator-runtime/runtime.js:202:11
      at callInvokeWithMethodAndArg (node_modules/regenerator-runtime/runtime.js:201:16)
      at AsyncIterator.enqueue (node_modules/regenerator-runtime/runtime.js:224:13)
      at AsyncIterator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at Object.runtime.async (node_modules/regenerator-runtime/runtime.js:248:14)
      at Context._callee (test/app/actions/login.test.js:13:16)

  2) Actions.Views.SelectServer handleServerUrlChanged:
     TypeError: Cannot read property 'isConnected' of undefined
      at networkConnectionListener (app/utils/network.js:34:2)
      at Object.detectNetwork (app/store/index.js:199:2)
      at configureServiceStore (node_modules/mattermost-redux/store/configureStore.dev.js:91:27)
      at configureAppStore (app/store/index.js:199:2)
      at Object._callee$ (test/app/actions/select_server.test.js:13:23)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at invoke (node_modules/regenerator-runtime/runtime.js:155:20)
      at node_modules/regenerator-runtime/runtime.js:202:11
      at callInvokeWithMethodAndArg (node_modules/regenerator-runtime/runtime.js:201:16)
      at AsyncIterator.enqueue (node_modules/regenerator-runtime/runtime.js:224:13)
      at AsyncIterator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at Object.runtime.async (node_modules/regenerator-runtime/runtime.js:248:14)
      at Context._callee (test/app/actions/select_server.test.js:12:34)

  3) Actions.Views.Thread handleCommentDraftChanged:
     TypeError: Cannot read property 'isConnected' of undefined
      at networkConnectionListener (app/utils/network.js:34:2)
      at Object.detectNetwork (app/store/index.js:199:2)
      at configureServiceStore (node_modules/mattermost-redux/store/configureStore.dev.js:91:27)
      at configureAppStore (app/store/index.js:199:2)
      at Object._callee$ (test/app/actions/thread.test.js:13:23)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at invoke (node_modules/regenerator-runtime/runtime.js:155:20)
      at node_modules/regenerator-runtime/runtime.js:202:11
      at callInvokeWithMethodAndArg (node_modules/regenerator-runtime/runtime.js:201:16)
      at AsyncIterator.enqueue (node_modules/regenerator-runtime/runtime.js:224:13)
      at AsyncIterator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at Object.runtime.async (node_modules/regenerator-runtime/runtime.js:248:14)
      at Context._callee (test/app/actions/thread.test.js:12:37)
```

**Comment by jarredwitt** [comment]
@enahum @hmhealey tests are passing now, just needed to mock NetInfo

File: app/utils/network.js (hunk 1 above)
Comment (jarredwitt) [reply app/utils/network.js]: Wanted to keep it all on one line. Eslint didn't like the function return value for the arrow function

**Pull request merged**